signing_service/src/signing_service.py

```diff
@@ -0,0 +1,139 @@
+import argparse
+import logging.config
+import os
+import signal
+import socket
+from time import sleep
+
+
+logger = logging.getLogger()
+crash_logger = logging.getLogger('crash')
+
+
+class SigningService:
+    """
+    The Signing Service connects to Middleman as a client but then listens for requests coming from Concent via
+    Middleman. The underlying protocol is TCP and data sent over that is expected to conform to the Wire protocol.
+    """
+
+    def __init__(self, host, port, reconnect_delay) -> None:
+        assert isinstance(host, str)
+        assert isinstance(port, int)
+        assert isinstance(reconnect_delay, int)
+        self.host = host
+        self.port = port
+        self.reconnect_delay = reconnect_delay
+        self.current_reconnect_delay = 0
+        self.socket: socket.socket = None  # type: ignore
+        self.is_connection_established = False
+
+        # Handle shutdown signal.
+        signal.signal(signal.SIGTERM, self._terminate_connection_and_exit)  # type: ignore
+
+        self._run()
```

Comment:
personaly, I would remove it from `__init__` and call it explicitly (`SigningService(arg_host, arg_port, arg_reconnect_delay).run()`)

Reply:
I can do this but could you provide some argumentation for this ?

Reply:
It is more visible, that you actually start the service. If start is automatic (you cannot create object without it), it is less flexible and might be misleading sometimes.

Reply:
Ok, sounds good, will do.

Reply:
:+1: done